GitUI/CommandsDialogs/BrowseDialog/FormUpdates.cs

```diff
@@ -123,6 +126,7 @@ private void Done()
                 {
                     UpdateLabel.Text = _noUpdatesFound.Text;
                 }
+                Dispose();
```

Comment:
I was trying to move the dispose from the FormBrowser to the FormUpdates itself there by avoiding access to a disposedobject. As you said, I should dispose it if there's no updates.

I'd go through the `FormUpdate` to see if any particular reason the form needs to be disposed manually. If not I'd just remove the dispose call, and leave it to the GC.